src/de/variant_de.rs

```diff
@@ -1,40 +1,36 @@
-use crate::safearray::get_string_array;
+use crate::safearray::safe_array_to_vec_of_strings;
+use crate::variant::Variant;
 use failure::{bail, Error};
 use log::debug;
 use serde::{de, forward_to_deserialize_any, Deserialize};
 use std::fmt;
-use crate::variant::Variant;
+use std::vec::IntoIter;
 
+#[derive(Debug)]
 struct SeqAccess {
-    data: Vec<Variant>,
-    i: usize,
+    data: IntoIter<Variant>,
 }
 
 impl<'de> de::SeqAccess<'de> for SeqAccess {
     type Error = crate::error::Error;
 
     fn next_element_seed<T>(&mut self, seed: T) -> Result<Option<T::Value>, Self::Error>
-        where
-            T: de::DeserializeSeed<'de>,
+    where
+        T: de::DeserializeSeed<'de>,
     {
-        if self.i >= self.data.len() {
-            return Ok(None);
+        match self.data.next() {
+            Some(variant) => seed.deserialize(variant).map(Some),
```

Comment:
This took me a moment to realize what's happening ?
One branch returns Some(deserialized), and the other returns Ok(None)

Unless `seed.deserialize` also returns a result, and than it wraps the inner type in Some, which would be 
Ok(Some(..)), but than it's very messy. 

Consider:
```
if let Some(variant) = self.data.next() {
    return Some(seed.sereialize(variant))
}

return Ok(None)
```

Reply:
It's the way `seed` is used normally. I'll make more clear :)

Reply:
But your code is incorrect, since you return an `Option<Result>`, and we need a `Result<Option>`